Fix: update Product.item components

madcalf opened the pull request:
Updates the `Product.info` components' field names to be more consistent and/or clarify their intent.
Also removes the default values in the `weight` component.

Closes #63

jtfairbank reviewed (changes requested):
* **Praise:** Great job touching up all of the documented consistency issues!
* **Suggestion (blocking):** Noticed a few others while I was reviewing.  Can we address them in this PR too?
    - for `item.needsMet`, rename `monthlyNeedsMetPerUnit` to `monthlyNeedsMetPerItem` so that we're using the item / unit terminology consistently
    - for `item.value` make the `source` and `logDate` fields required, to be consistent with `item.volume` and `item.weight`
* **Question:** For `item.value`, do you think it'd be worth adding a `packagePriceUnit` field, set to `USD` as a constant for now?  And what about naming `itemPrice` to `itemPriceUSD` for consistency with `itemVolume` and `itemWeight` which include units in the field name?

jtfairbank commented:
@madcalf testing 123

madcalf commented:
> * **Suggestion (blocking):** Noticed a few others while I was reviewing.  Can we address them in this PR too?
>   
>   * for `item.needsMet`, rename `monthlyNeedsMetPerUnit` to `monthlyNeedsMetPerItem` so that we're using the item / unit terminology consistently
>   * for `item.value` make the `source` and `logDate` fields required, to be consistent with `item.volume` and `item.weight`

> For `item.value`, do you think it'd be worth adding a `packagePriceUnit` field, set to `USD` as a constant for now?  

Added `packagePriceUnit` field as an enum with only the `USD` option and set `USD` as the default value, so we can add in additional currency values needed. 

> And what about naming `itemPrice` to `itemPriceUSD` for consistency with `itemVolume` and `itemWeight` which include units in the field name?

Renamed itemPrice to `pricePerItemUSD`. Assuming we want this to be a calculation

madcalf commented:
> rename `monthlyNeedsMetPerUnit` to `monthlyNeedsMetPerItem` so that we're using the item / unit terminology consistently

Rename is done. Will make `monthlyNeedsMetPerItem` a calculated field in a separate PR.

madcalf commented:
> For `item.value`, do you think it'd be worth adding a `packagePriceUnit` field, set to `USD` as a constant for now?  

Sure that makes sense. Added it as an enum with `USD` as the default (and currently the only) option, and we can add in additional currency values needed.

madcalf commented:
> And what about naming `itemPrice` to `itemPriceUSD` for consistency with `itemVolume` and `itemWeight` which include units in the field name?

Renamed itemPrice to `pricePerItemUSD` because i kept getting it confused with the `packagePrice` and it seems more in line with the other calculated field names. I'll also make this one a calculated field in a separate PR.

jtfairbank commented:
@madcalf looks great!
